# 20.440 Course Project - A network motif-enrichment approach to mapping genetic dysregulation in Amyotrophic Lateral Sclerosis

We analyze 21 tissue samples from the NeuroLINCS project to investigate the regulatory mechanisms underlying sporadic amyotrophic lateral sclerosis (ALS). Using a combination of regulatory network reconstruction, differential gene expression analysis, and network motif enrichment analysis, we identify and highlight 14 genes and three pathways which are perturbed in ALS, spanning motor neuron development, myosin binding, and oxidative stress. Our multi-pronged approach considers multiple sources of variation in disease and combines orthogonal sources of information to untangle complicated regulatory dynamics and identify sources of genetic dysregulation.

Before running this repository, please view our [report](https://github.com/samsledje/als_regulation_20440/blob/main/docs/Sledzieski_Kabani_ALS_Final_Report.pdf) and [slides](https://github.com/samsledje/als_regulation_20440/blob/main/docs/Sledzieski_Kabani_ALS_Final_Slides.pdf).

## Environment Setup

We have provided an `environment.yml` file which contains the environment necessary to run the scripts in this notebook. The conda environment can be created using the command

```
conda env create -n ALS_regulation --file environment.yml
```

## Data

Our data for this study comes from the [NeuroLINCS](http://neurolincs.org) project, from which we use transcriptomic data from  9 control and 12 ALS iPSC tissue samples

## Software

This study uses several open-source software tools to perform our analysis:

* [DESeq2](https://bioconductor.org/packages/release/bioc/html/DESeq2.html)
* [GENIE3](https://bioconductor.org/packages/release/bioc/html/GENIE3.html)
* [GSEA](https://www.gsea-msigdb.org)
* [mFinder](https://www.weizmann.ac.il/mcb/UriAlon/download/network-motif-software)

## Repository Organization

This repository contains five primary directories:

* `nb`, which contains all source code for this analysis in the form of Jupyter notebooks. Notebooks are further organized into
  * `iPSC`, which contains notebooks for analyzing iPSC tissue samples and
  * `iMN`, which contains essentially the same analysis, slightly modified to read and write our iMN specific data
* `data`, which contains raw data downloaded from NeuroLINCS (as well as a list of human transcription factors from [Lambert et al.](http://humantfs.ccbr.utoronto.ca))
* `nbdata.zip`, a zip archive containing all intermediate files processed or generated by our source code. You can unzip this archive and create the `nbdata` directory to immediately access all of those files, or you can create an empty `nbdata` directory and run the notebooks to generate these files yourself. **NOTE:** Even the zipped `nbdata` is too large for Git-LFS -- please contact the author (samsl@mit.edu) if you require access to the zipped intermediate files.
* `figs`, which contains all figures generated by the notebook and used in our report and presentation
* `docs`, which contains our report and slides
